OpenRA.Mods.Common/Traits/Air/Aircraft.cs

```diff
@@ -200,6 +200,7 @@ public virtual WVec GetRepulsionForce()
 
 			var repulsionForce = self.World.FindActorsInCircle(self.CenterPosition, Info.IdealSeparation)
 				.Where(a => !a.IsDead && a.Info.HasTraitInfo<AircraftInfo>()
+					&& a.Info.TraitInfo<AircraftInfo>().Repulsable
 					&& a.Info.TraitInfo<AircraftInfo>().CruiseAltitude == Info.CruiseAltitude)
```

Comment:
This now does three queries for `AircraftInfo`, which isn't very efficient.  Can you change this to something like the following?

```csharp

.Where(a =>
{
    if (a.IsDead)
        return false;
    var ai = a.Info.TraitInfoOrDefault<AircraftInfo>();
    return ai != null && ai.Repulsable && ai.CruiseAltitude == Info.CruiseAltitude;
})
```